src/main/java/pl/jagiellonian/model/PolynomialAsListOfMaps.java

```diff
@@ -0,0 +1,242 @@
+package pl.jagiellonian.model;
+
+import pl.jagiellonian.exceptions.WrongFormatException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+public class PolynomialAsListOfMaps {
+    private final static String CONSTANT = "\\(?(-?[1-9][0-9]*)\\)?\\*?";
+    private final static Pattern CONSTANT_IN_EXPRESSION = Pattern.compile("(?:" + CONSTANT + ")?" + "(.*)");
+    private final static String VARIABLE = "\\*?(?:([a-zA-Z]+_?[0-9]*)(?:\\^(-?[0-9]+))?)";
+    private final static Pattern VARIABLE_IN_EXPRESSION = Pattern.compile(VARIABLE);
+    private final static String VARIABLES = format("%s(?:\\*%s)*", VARIABLE, VARIABLE);
+    private final static Pattern VALID_EXPRESSION = Pattern.compile(format("(%s{1}(?:%s)*|(?:%s)+)(?:\\+(%s{1}(?:%s)*|(?:%s)+))*", CONSTANT, VARIABLES, VARIABLES, CONSTANT, VARIABLES, VARIABLES));
+
+    protected List<SingleExpression> expressions = new ArrayList<>();
+
+    private PolynomialAsListOfMaps() {
+    }
+
+    private PolynomialAsListOfMaps(SingleExpression expressions) {
+        this.expressions.add(expressions);
+    }
+
+    public PolynomialAsListOfMaps(List<SingleExpression> expressions) {
+        this.expressions = new ArrayList<>(expressions);
+    }
+
+    public int size() {
+        return expressions.size();
+    }
+
+    public List<SingleExpression> getExpressions() {
+        return new ArrayList<>(expressions);
+    }
+
+    public static PolynomialAsListOfMaps parse(String expression) {
+        expression = expression.replaceAll("\\s+", "");
+        Matcher matcher = VALID_EXPRESSION.matcher(expression);
+        if (!matcher.matches()) {
+            throw new WrongFormatException();
+        }
+        String[] singleExpressions = expression.split("\\+");
+        PolynomialAsListOfMaps result = new PolynomialAsListOfMaps();
+        for (String singleExpressionString : singleExpressions) {
+            SingleExpression singleExpression = parseSingleExpression(singleExpressionString);
+            if (singleExpression != null) {
+                result.expressions.add(singleExpression);
+            }
+        }
+        result.simplifyExpressions();
+        return result;
+    }
+
+    private static SingleExpression parseSingleExpression(String singleExpressionString) {
+        Matcher constantMatcher = CONSTANT_IN_EXPRESSION.matcher(singleExpressionString);
+        constantMatcher.matches();
```

Comment:
Result is ignored because it's already checked(At least now when this method is used only from parse method which have already checked if expression String is correct)